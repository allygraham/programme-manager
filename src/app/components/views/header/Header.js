import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'react-emotion';

import { constants } from '~/constants';

const Title = styled('h1')`
  color: #FFF;
`;

const LinkText = styled('span')`
  margin-left: 1rem;
`;

export const Header = () => (
  <nav className="header navbar is-link">
    <div className="navbar-brand">
      <Link to={constants.routes.index}>
        <Title className="navbar-item is-size-5">
          {constants.titles.app}
        </Title>
      </Link>
    </div>
    <div className="navbar-menu is-active">
      <div className="navbar-start">
        <Link to={constants.routes.programmeList} className="navbar-item">
          <i className="fa fa-list"/>
          <LinkText>{constants.titles.programmeList}</LinkText>
        </Link>
      </div>
    </div>
  </nav>
);
