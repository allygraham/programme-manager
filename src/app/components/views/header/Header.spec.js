import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header', () => {

  test('render the <Header> component', () => {
    const wrapper = shallow(<Header/>);
    expect(wrapper).toMatchSnapshot();
  });

});
