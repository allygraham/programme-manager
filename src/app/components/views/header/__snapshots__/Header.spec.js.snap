// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render the <Header> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="navbar-brand"
        >
          <Link
            replace={false}
            to="/home"
          >
            <Styled(h1)
              className="navbar-item is-size-5"
            >
              Programme Manager
            </Styled(h1)>
          </Link>
        </div>,
        <div
          className="navbar-menu is-active"
        >
          <div
            className="navbar-start"
          >
            <Link
              className="navbar-item"
              replace={false}
              to="/programme-list"
            >
              <i
                className="fa fa-list"
              />
              <Styled(span)>
                Programme List
              </Styled(span)>
            </Link>
          </div>
        </div>,
      ],
      "className": "header navbar is-link",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/home"
          >
            <Styled(h1)
              className="navbar-item is-size-5"
            >
              Programme Manager
            </Styled(h1)>
          </Link>,
          "className": "navbar-brand",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(h1)
              className="navbar-item is-size-5"
            >
              Programme Manager
            </Styled(h1)>,
            "replace": false,
            "to": "/home",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Programme Manager",
              "className": "navbar-item is-size-5",
            },
            "ref": null,
            "rendered": "Programme Manager",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="navbar-start"
          >
            <Link
              className="navbar-item"
              replace={false}
              to="/programme-list"
            >
              <i
                className="fa fa-list"
              />
              <Styled(span)>
                Programme List
              </Styled(span)>
            </Link>
          </div>,
          "className": "navbar-menu is-active",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="navbar-item"
              replace={false}
              to="/programme-list"
            >
              <i
                className="fa fa-list"
              />
              <Styled(span)>
                Programme List
              </Styled(span)>
            </Link>,
            "className": "navbar-start",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="fa fa-list"
                />,
                <Styled(span)>
                  Programme List
                </Styled(span)>,
              ],
              "className": "navbar-item",
              "replace": false,
              "to": "/programme-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fa fa-list",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Programme List",
                },
                "ref": null,
                "rendered": "Programme List",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="navbar-brand"
          >
            <Link
              replace={false}
              to="/home"
            >
              <Styled(h1)
                className="navbar-item is-size-5"
              >
                Programme Manager
              </Styled(h1)>
            </Link>
          </div>,
          <div
            className="navbar-menu is-active"
          >
            <div
              className="navbar-start"
            >
              <Link
                className="navbar-item"
                replace={false}
                to="/programme-list"
              >
                <i
                  className="fa fa-list"
                />
                <Styled(span)>
                  Programme List
                </Styled(span)>
              </Link>
            </div>
          </div>,
        ],
        "className": "header navbar is-link",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/home"
            >
              <Styled(h1)
                className="navbar-item is-size-5"
              >
                Programme Manager
              </Styled(h1)>
            </Link>,
            "className": "navbar-brand",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(h1)
                className="navbar-item is-size-5"
              >
                Programme Manager
              </Styled(h1)>,
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Programme Manager",
                "className": "navbar-item is-size-5",
              },
              "ref": null,
              "rendered": "Programme Manager",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="navbar-start"
            >
              <Link
                className="navbar-item"
                replace={false}
                to="/programme-list"
              >
                <i
                  className="fa fa-list"
                />
                <Styled(span)>
                  Programme List
                </Styled(span)>
              </Link>
            </div>,
            "className": "navbar-menu is-active",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="navbar-item"
                replace={false}
                to="/programme-list"
              >
                <i
                  className="fa fa-list"
                />
                <Styled(span)>
                  Programme List
                </Styled(span)>
              </Link>,
              "className": "navbar-start",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="fa fa-list"
                  />,
                  <Styled(span)>
                    Programme List
                  </Styled(span)>,
                ],
                "className": "navbar-item",
                "replace": false,
                "to": "/programme-list",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fa fa-list",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Programme List",
                  },
                  "ref": null,
                  "rendered": "Programme List",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
